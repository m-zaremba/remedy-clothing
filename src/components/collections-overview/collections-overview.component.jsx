import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CollectionPreview from '../collection-preview/collection-preview.component';

import { selectCollectionsForPreview } from '../../redux/shop/shop.selectors';

import { CollectionsOverviewContainer } from './collections-overview.styles';

const CollectionsOverview = ({ collections }) => {
  return (
    <CollectionsOverviewContainer>
      {collections.map(({ id, title, items, history, match, routeName }) => (
        <CollectionPreview
          key={id}
          title={title}
          items={items}
          history={history}
          match={match}
          routeName={routeName}
        />
      ))}
    </CollectionsOverviewContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  collections: selectCollectionsForPreview,
});

CollectionsOverview.propTypes = {
  collections: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default connect(mapStateToProps)(CollectionsOverview);
