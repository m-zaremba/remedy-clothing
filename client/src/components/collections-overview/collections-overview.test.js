import React from 'react';
import { shallow } from 'enzyme';
import { CollectionsOverview } from './collections-overview.component';

describe('testing CollectionsOverview component', () => {
  it('should render CollectionsOverview component', () => {
    expect(shallow(<CollectionsOverview collections={[]} />)).toMatchSnapshot();
  });
});
