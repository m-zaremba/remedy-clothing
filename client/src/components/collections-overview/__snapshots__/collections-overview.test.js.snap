// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing CollectionsOverview component should render CollectionsOverview component 1`] = `<styled.div />`;
